project/src/components/movie-page-details-screen/movie-page-details-screen.tsx

```diff
@@ -1,22 +1,34 @@
 
-import { Link} from 'react-router-dom';
-import { AppRoute} from '../../consts';
+import { Link, useParams} from 'react-router-dom';
+import { AppRoute, AuthorizationStatus} from '../../consts';
 import NotFound from '../404-screen/404-screen';
-import { useFilm } from '../../hooks/use-film';
 import Tabs from '../tabs/tabs';
 import LikeFilms from '../like-films/like-films';
 import Header from '../header/header';
 import Footer from '../footer/footer';
+import { useEffect } from 'react';
+import { fetchCurrentFilmAction, fetchMoreLikeFilmsAction, fetchReviewsAction } from '../../store/api-actions';
+import { store } from '../../store';
+import { useAppSelector } from '../../hooks';
+import { Film } from '../../types/film';
 
 function MovieDetails(): JSX.Element {
-  const film = useFilm();
-
-
-  if(typeof film === 'undefined'){
+  const params = useParams();
+  useEffect(() => {
+    store.dispatch(fetchCurrentFilmAction(params.id as string));
+    store.dispatch(fetchMoreLikeFilmsAction(params.id as string));
+    store.dispatch(fetchReviewsAction(params.id as string));
+  }, [params.id]);
```

Comment:
Вариант работает (возможно)
дело в том, что можно нарваться на циркуляную зависимость модулей
специально для этого есть крючок 
```suggestion
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCurrentFilmAction(params.id as string));
    dispatch(fetchMoreLikeFilmsAction(params.id as string));
    dispatch(fetchReviewsAction(params.id as string));
  }, [params.id]);
```

Надо исправить

Reply:
И опять множественные вызовы dispatch